pkg/helper/options/delete.go

```diff
@@ -31,9 +31,13 @@ type DeleteOptions struct {
 	DeleteRelated      bool
 	DeleteAllNs        bool
 	DeleteAll          bool
+	Keep               int
 }
 
 func (o *DeleteOptions) CheckOptions(s *cli.Stream, resourceNames []string, ns string) error {
+	if o.Keep > 0 && !(o.DeleteAllNs || o.DeleteAll) {
```

Comment:
Promise this is the last thing. Could the error message be:

```suggestion
		return fmt.Errorf("must use --all flag with --keep")
```